### PR TITLE
[pulsar-client] add pending-queue size metrics to producer stats

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation
Add pending-queue size metrics to producer stats to create monitoring at client side and monitor various scenarios : timeout, high latency, memory, etc..

### Modification 
Add API `ProducerStats::getPendingQueueSize` to retrieve pending-queue size metrics for a producer.

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerStatTest.java [hunk 1]
```diff
@@ -424,4 +424,22 @@ public void testAddBrokerLatencyStats() throws Exception {
...
         log.info("-- Exiting {} test --", methodName);
     }
+
+    @Test
+    public void testProducerPendingQueueSizeStats() throws Exception {
+        log.info("-- Starting {} test --", methodName);
+        ProducerBuilder<byte[]> producerBuilder = pulsarClient.newProducer()
+                .topic("persistent://my-property/tp1/my-ns/my-topic1");
+
+        Producer<byte[]> producer = producerBuilder.enableBatching(false).create();
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerStatTest.java:434]: What about adding @Cleanup ?

**Review by eolivelli — APPROVED** [review]
Lgtm

File: pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestRunMain.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
     public void runMainNoArguments() throws Exception {
         PulsarAdminTool.setAllowSystemExit(false);
         PulsarAdminTool.main(new String[0]);
-        assertEquals(PulsarAdminTool.getLastExitCode(), 1);
+        assertEquals(PulsarAdminTool.getLastExitCode(), 0);
```
Comment (eolivelli) [inline pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestRunMain.java:34]: This change looks unrelated

Comment (rdhabalia) [reply pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestRunMain.java]: yes, it was for testing. created a separate PR: https://github.com/apache/pulsar/pull/12675

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerStatTest.java (hunk 1 above)
Comment (rdhabalia) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerStatTest.java:437]: this test closes `pulsarClient` after every test method, which will also close and clean up producers/consumers. However, I just added the cleanup annotation.

**Review by codelipenghui — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM
